File: ffmpegdemo/testffmpeg/src/main/java/aplay/testffmpeg/MainActivity.java
package aplay.testffmpeg;

import android.app.Activity;
import android.os.Bundle;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    // Used to load the 'native-lib' library on application startup.
    static {
        System.loadLibrary( "native-lib" );
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate( savedInstanceState );
        setContentView( R.layout.activity_main );

        // Example of a call to a native method
        TextView tv = (TextView) findViewById( R.id.sample_text );
        Open( "/sdcard/1080.mp4",this );
        tv.setText( stringFromJNI() );
    }

    /**
     * A native method that is implemented by the 'native-lib' native library,
     * which is packaged with this application.
     */
    public native String stringFromJNI();
    public native boolean Open(String url,Object handle);
}

File: android/RouterProject/module_1/src/main/java/com/joyy/module_1/MainActivity.kt
package com.joyy.module_1

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.com.joyy.module_1.R
import com.joyy.router.annotations.Destination

@Destination(
    url = "module_1/MainActivity",
    description = "MainActiity"
)
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }
}